### PR TITLE
sound environments

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request

VIDEO: https://www.youtube.com/watch?v=M3tli1qg7jY

Implements functionality for byond's native sound environments feature.
Allows effects on sounds based on the surrounding area.
Currently uses a couple of different presets based on the area the person being played to is in.

Downsides:
sound environments are global to every single sound being played and this will affect _every_ sound including things like the bwoink.

## Why It's Good For The Game

Makes a huge difference in immersion and believability for such a tiny change

## Changelog
:cl:
soundadd: sound environments
/:cl:

### CODE REVIEW

**Comment by Shapsy** [comment]
fully immersive echo bwoinks

**Comment by Akrilla** [comment]
I think I've seen this on other servers, if it's the same thing then this is a really nice change

File: code/game/area/areas/centcom.dm [hunk 1]
```diff
@@ -108,6 +110,7 @@
 	has_gravity = STANDARD_GRAVITY
 	noteleport = TRUE
 	flags_1 = NONE
+	sound_environment
```
Comment (ike709) [inline code/game/area/areas/centcom.dm]: 🤔

**Comment by mattroks101** [comment]
It would be a good idea to implement defines for these environment sounds like what Bay has, so people looking at the code have a good idea of what it's going to sound like.
image

**Comment by actioninja** [comment]
I did but the fucking PR isn't updating

**Comment by actioninja** [comment]
update you shit

**Comment by TheSilverNuke** [comment]
Very cool

**Comment by SpaceManiac** [comment]
will have to play with this for a while but so far everything just sounds echoey

**Comment by AffectedArc07** [comment]
Or just stick a VST on your output

**Comment by Fox-McCloud** [comment]
Are we really going here again?

There's literally no way to fix the bugs with this, as environments are applied globally, and not on a per sound basis.

You can do what bay does and hack it/cheat, by resetting the sound environment, playing the sound in the current environment (based on area), but this doesn't cut out the problem.

This invariably means that sounds played close together where there isn't environmental setting will inherit the environment var from the previously played sound; it also means that you'll get weird cases of Lavaland suddenly echoing like it's a bathroom, or an adminhelp being really really dull/quiet one time, but a super long drawn out echo the next.

This has been submitted multiple times and tried multiple times, but there's no reliable way around the bug, and you will always have problems with this, because of how `environment` is in BYOND---it seems to be built around the assumption that it'll either be a single player game or that you won't be using sounds being pushed out to multiple clients, close together, in vastly disparate areas.

**Comment by SpaceManiac** [comment]
@actioninja you have reopen powers if you figure out how to solve the above

**Pull request closed without merging**